sonarts-core/src/rules/argumentsOrderRule.ts

```diff
@@ -0,0 +1,137 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { TypedSonarRuleVisitor, IssueLocation } from "../utils/sonarAnalysis";
+import { isIdentifier } from "../utils/nodes";
+
+export class Rule extends tslint.Rules.TypedRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "arguments-order",
+    description: "Parameters should be passed in the correct order",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-2234",
+    type: "functionality",
+    typescriptOnly: false,
+  };
+
+  public static MESSAGE = `Parameters have the same names but not the same order as the arguments.`;
+
+  public applyWithProgram(sourceFile: ts.SourceFile, program: ts.Program): tslint.RuleFailure[] {
+    return new Visitor(this.getOptions().ruleName, program).visit(sourceFile).getIssues();
+  }
+}
+
+class Visitor extends TypedSonarRuleVisitor {
+  visitCallExpression(node: ts.CallExpression) {
+    const argumentNames = this.getArgumentNames(node);
+    const { parameterNames, declaration } = this.getSignature(node);
+
+    if (parameterNames) {
+      const hasProblem = argumentNames.some((argumentName, argumentIndex) => {
+        if (!argumentName) {
+          return false;
+        }
+        return this.isArgumentSwapped(argumentName, argumentIndex, argumentNames, parameterNames, node);
+      });
+
+      if (hasProblem) {
+        this.raiseIssue(node, declaration);
+      }
+    }
+    super.visitCallExpression(node);
+  }
+
+  isArgumentSwapped(
+    argumentName: string,
+    argumentIndex: number,
+    argumentNames: (string | undefined)[],
+    parameterNames: ts.Symbol[],
+    node: ts.CallExpression,
+  ) {
+    const parameterIndex = parameterNames.findIndex(param => param.name === argumentName);
+    if (parameterIndex > -1 && parameterIndex !== argumentIndex) {
+      const anotherArgument = argumentNames[parameterIndex];
+      if (parameterNames[argumentIndex].name === anotherArgument) {
+        const firstArgumentType = this.getTypeAsString(node.arguments[argumentIndex]);
+        const secondArgumentType = this.getTypeAsString(node.arguments[parameterIndex]);
+
+        if (firstArgumentType === secondArgumentType) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  getTypeAsString(expr: ts.Expression) {
+    const typeChecker = this.program.getTypeChecker();
```

Comment:
Nitpicking: 
You could use destructuring for `typeToString`, `getBaseTypeOfLiteralType` and `getTypeAtLocation ` functions from typeChecker